firefox-ios/Client/Frontend/Home/Homepage Rebuild/Redux/HomepageState.swift

```diff
@@ -63,11 +59,24 @@ struct HomepageState: ScreenState, Equatable {
                 pocketState: PocketState.reducer(state.pocketState, action)
             )
         default:
-            return HomepageState(
-                windowUUID: state.windowUUID,
-                headerState: HeaderState.reducer(state.headerState, action),
-                pocketState: PocketState.reducer(state.pocketState, action)
-            )
+            return defaultActionState(from: state, action: action)
         }
     }
+
+    private static func defaultActionState(from state: HomepageState, action: Action) -> HomepageState {
+        let defaultState = defaultActionState(from: state)
+        return HomepageState(windowUUID: defaultState.windowUUID,
+                             headerState: HeaderState.reducer(state.headerState, action),
+                             pocketState: PocketState.reducer(state.pocketState, action))
+    }
+
+    static func defaultActionState(from state: HomepageState) -> HomepageState {
+        // since the default state would depend on other reducers use `defaultActionState(for: ,action:)`
+        // as method for default action state
+        return HomepageState(
+            windowUUID: state.windowUUID,
+            headerState: state.headerState,
```

Comment:
I think we should call the child reducers here - they should return their own default state in this case anyway and with that we wouldn't need two methods.